crates/sui-types/src/transaction.rs

```diff
@@ -113,6 +116,22 @@ pub enum ObjectArg {
     Receiving(ObjectRef),
 }
 
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize)]
+pub struct BalanceWithdrawArg {
+    /// The maximum amount of the balance to withdraw.
+    /// If None, reserve the entire balance.
+    pub max_amount: Option<u64>,
+    /// The source of the balance to withdraw.
+    pub withdraw_from: WithdrawFrom,
+}
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize)]
+pub enum WithdrawFrom {
+    /// Withdraw from the sender of the transaction, with balance type T as T in Balance<T>.
+    Sender(TypeTag),
```

Comment:
This should be `TypeInput` and I think we should move the `TypeInput` onto the arg itself. I don't really see a case where the type specifier wont' be necessary

Reply:
(also TypeInput not TypeTag)

Reply:
Why do we need TypeInput now?

Reply:
I noticed that we are still using TypeTag everywhere else though. I added a TODO there to make sure we update this before we ever ship it.

Reply:
Yeah for some reasons, TypeInput is used in (most/all?) places inside of a TransactionData, everywhere else it is a TypeTag. But i wonder if we need to use a TypeInput here given this is a newly added field without the previous baggage